fuel-core/src/schema/dap.rs

```diff
@@ -192,4 +192,272 @@ impl DapMutation {
+            Ok(self::gql_types::RunResult {
+                state: match dbgref {
+                    Some(_) => self::gql_types::RunState::Breakpoint,
+                    None => self::gql_types::RunState::Completed,
+                },
+                breakpoint: dbgref.and_then(|d| match d {
+                    DebugEval::Continue => None,
+                    DebugEval::Breakpoint(bp) => Some(bp.into()),
+                }),
+            })
+        }
+
+        #[cfg(not(feature = "debug"))]
+        {
+            let _ = state_ref;
+            Ok(self::gql_types::RunResult {
+                state: self::gql_types::RunState::Completed,
+                breakpoint: None,
+            })
+        }
+    }
+
+    #[cfg(not(feature = "debug"))]
+    async fn continue_tx(
+        &self,
+        _ctx: &Context<'_>,
+        _id: ID,
+    ) -> async_graphql::Result<self::gql_types::RunResult> {
+        Err(async_graphql::Error::new(
+            "Feature 'debug' is not compiled in",
+        ))
+    }
+
+    #[cfg(feature = "debug")]
+    async fn continue_tx(
+        &self,
+        ctx: &Context<'_>,
+        id: ID,
+    ) -> async_graphql::Result<self::gql_types::RunResult> {
+        trace!("Continue exectuon of VM {:?}", id);
+
+        let mut locked = ctx.data_unchecked::<GraphStorage>().lock().await;
+        let vm = locked
+            .vm
+            .get_mut(&id)
+            .ok_or_else(|| async_graphql::Error::new("VM not found"))?;
+        let state = match vm.resume() {
+            Ok(state) => state,
+            // The transaction was already completed earlier, so it cannot be resumed
+            Err(fuel_vm::error::InterpreterError::DebugStateNotInitialized) => {
+                return Ok(self::gql_types::RunResult {
+                    state: self::gql_types::RunState::Completed,
+                    breakpoint: None,
+                })
+            }
+            // The transaction was already completed earlier, so it cannot be resumed
+            Err(err) => return Err(async_graphql::Error::new(format!("VM error: {err:?}"))),
+        };
+
+        let dbgref = state.debug_ref();
+
+        Ok(self::gql_types::RunResult {
+            state: match dbgref {
+                Some(_) => self::gql_types::RunState::Breakpoint,
+                None => self::gql_types::RunState::Completed,
+            },
+            breakpoint: dbgref.and_then(|d| match d {
+                DebugEval::Continue => None,
+                DebugEval::Breakpoint(bp) => Some(bp.into()),
+            }),
+        })
+    }
+}
+
+mod gql_types {
+    //! GraphQL type wrappers
+    use async_graphql::*;
+
+    use fuel_types::Word;
+    #[cfg(feature = "debug")]
+    use fuel_vm::prelude::Breakpoint as FuelBreakpoint;
+
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, InputObject)]
+    pub struct ContractId {
+        value: [u8; 32],
+    }
+    impl From<&fuel_types::ContractId> for ContractId {
```

Comment:
Since we deref, does it make sense to have an additional `From` here? This will give the impression to the user that we use a transformed type internally - when in fact, we just copy & mirror the owned `From`

Reply:
That's a good point